python/pyspark/sql/pandas/group_ops.py

```diff
@@ -463,12 +459,20 @@ def applyInPandas(
 
         # The usage of the pandas_udf is internal so type checking is disabled.
         udf = pandas_udf(
-            func, returnType=schema, functionType=PythonEvalType.SQL_COGROUPED_MAP_PANDAS_UDF
+            func,
+            returnType=schema,
+            functionType=PythonEvalType.SQL_COGROUPED_MAP_PANDAS_UDF,
         )  # type: ignore[call-overload]
 
-        all_cols = self._extract_cols(self._gd1) + self._extract_cols(self._gd2)
-        udf_column = udf(*all_cols)
-        jdf = self._gd1._jgd.flatMapCoGroupsInPandas(self._gd2._jgd, udf_column._jc.expr())
+        all_cols = self._extract_cols(self._gd1)
+        for gd in self._gds:
+            all_cols.extend(self._extract_cols(gd))
```

Comment:
Could be simplified to
```suggestion
        all_cols = [self._extract_cols(gd) for gd in [self._gd1] + self._gds]
```